Hook up label to input for dontShowAgain checkbox

cstyles opened the pull request:
[//]: # (Note: See http://habitica.fandom.com/wiki/Using_Your_Local_Install_to_Modify_Habitica%27s_Website_and_API for more info)

[//]: # (Put Issue # here, if applicable. This will automatically close the issue if your PR is merged in)

### Changes
[//]: # (Describe the changes that were made in detail here. Include pictures if necessary)
Now if the user clicks on the label for the checkbox, it will (un)check the associated checkbox.


[//]: # (Put User ID in here - found on the Habitica website at User Icon > Settings > API)

----
UUID: `67b7e848-03f4-48af-b87b-6a715a729730`

paglias commented on website/client/src/components/achievements/streak.vue:
```diff
@@ -29,12 +29,14 @@
-            @change="suppressModals"
-          >
-          <label>{{ $t('dontShowAgain') }}</label>
+          <label>
```
thanks @cstyles , I think that for accessibility reasons it would be better to keep the label and input as siblings while adding an unique `id` attribute to the input and a matching `for` attribute to the label meaning that the label is linked to the input when using a screen readeer, etc...

Putting the input inside the labels works as well but the w3 site suggests using explicit labels (https://www.w3.org/WAI/tutorials/forms/labels/#associating-labels-implicitly) and Bootstrap's styles work better with the label used as a sibling of the input (https://getbootstrap.com/docs/4.3/components/forms/)

cstyles replied:
Yeah, I would have preferred that too but as I'm not familiar with the codebase I didn't know what to pick to avoid conflicts. Would something like `dontShowAgain` suffice?

paglias replied:
I would probably use something that matches the user preference so in this case `user-preference-suppressModals-streak`

cstyles replied:
OK, I just pushed the change. Let me know what you think!

cstyles committed:
Hook up label to input for dontShowAgain checkbox

Now if the user clicks on the label for the checkbox, it will (un)check
the associated checkbox.

paglias commented:
Sorry for the delay @cstyles ! This has been merged and will go live next week, I've notated the PR towards your 2nd contributor tier
